tomviz/CentralWidget.cxx

```diff
@@ -432,7 +437,12 @@ void CentralWidget::setDataSource(DataSource* source)
 
 void CentralWidget::onColorMapUpdated()
 {
-  this->refreshHistogram();
+  this->onDataSourceChanged();
+}
+
+void CentralWidget::onDataSourceChanged()
+{
+  this->Internals->Timer.start();
```

Comment:
Might be useful to add a comment explaining this for those not knowing the API so well
// Start (or restart) the one shot timer for delayed histogram refresh.